src/sidebar/components/search-status-bar.js

```diff
@@ -1,63 +1,109 @@
 'use strict';
 
-const memoize = require('../util/memoize');
+const { Fragment, createElement } = require('preact');
+const propTypes = require('prop-types');
 
+const { useMemo } = require('preact/hooks');
+const { withServices } = require('../util/service-context');
 const uiConstants = require('../ui-constants');
+const useStore = require('../store/use-store');
 
-// @ngInject
-function SearchStatusBarController(store, rootThread) {
-  this.TAB_ANNOTATIONS = uiConstants.TAB_ANNOTATIONS;
-  this.TAB_NOTES = uiConstants.TAB_NOTES;
-  this.TAB_ORPHANS = uiConstants.TAB_ORPHANS;
+function SearchStatusBar({
+  selectedTab,
+  totalAnnotations,
+  totalNotes,
+  rootThread,
+}) {
+  const storeState = useStore(store => store.getState());
+  const onClearSelection = useStore(store => store.clearSelection);
+  const filterQuery = storeState.filterQuery;
+  const filterActive = !!storeState.filterQuery;
 
-  const thread = () => {
-    return rootThread.thread(store.getState());
-  };
+  const thread = rootThread.thread(storeState);
 
-  const visibleCount = memoize(thread => {
-    return thread.children.reduce(
+  const countVisibleAnns = annThread => {
+    return annThread.children.reduce(
       function(count, child) {
-        return count + visibleCount(child);
+        return count + countVisibleAnns(child);
       },
-      thread.visible ? 1 : 0
+      annThread.visible ? 1 : 0
     );
-  });
+  };
+
+  const visibleCount = useMemo(() => {
+    return countVisibleAnns(thread);
+  }, [thread]);
 
-  this.filterMatchCount = function() {
-    return visibleCount(thread());
+  const filterResults = () => {
+    const resultsCount = visibleCount;
+    switch (resultsCount) {
+      case 0:
+        return 'No results for ' + filterQuery;
```

Comment:
There is a presumably unintended change to the display when the result count is 0. On master the filter query is quoted. In this branch it is not:

**Master:**

<img width="298" alt="Screenshot 2019-07-22 19 07 17" src="https://user-images.githubusercontent.com/2458/61654311-0afdb280-acb4-11e9-9d07-c4a559994302.png">


**This branch:**

<img width="340" alt="Screenshot 2019-07-22 19 07 22" src="https://user-images.githubusercontent.com/2458/61654287-ff11f080-acb3-11e9-9194-461c39f7624a.png">